lib/authHandlers.js

```diff
@@ -149,6 +151,25 @@ var sessions       = require('client-sessions'),
     }
   }
 
+  var handleSSOSetup = function(req, res, next) {
+    console.log("EREZ handleSSOSetup: ", req.query.access_token);
+    secondarySessionTokens[req.query.user_token] = req.query.access_token;
+    next();
+  }
+
+  var handleSSOLogin = function(req, res, next) {
+    console.log("EREZ handleSSOLogin: req.query = ", req.query);
+    if (req.query.user_token) {
+      console.log("EREZ handleSSOLogin: req has user token!!");
+      req.query.access_token = secondarySessionTokens[req.query.user_token];
```

Comment:
> What are the implications of not having a session token in the cache?

if ManageIQ did not insert a new session token to the cache ~immediately before the user starts a new session, it means the user is not authorized. This is why it is safe to remove the token from the cache once it's consumed by the user.